django/db/models/base.py

```diff
@@ -1283,7 +1284,13 @@ async def adelete(self, using=None, keep_parents=False):
 
     def _get_FIELD_display(self, field):
         value = getattr(self, field.attname)
-        choices_dict = dict(make_hashable(field.flatchoices))
```

Comment:
Could we instead have a code similar to this inside `GeneratedField` model definition? something like (untested):

```diff
diff --git a/django/db/models/fields/generated.py b/django/db/models/fields/generated.py
index 5b6b188df0..ffbed99f24 100644
--- a/django/db/models/fields/generated.py
+++ b/django/db/models/fields/generated.py
@@ -30,7 +30,7 @@ class GeneratedField(Field):
         self.expression = expression
         self.output_field = output_field
         self.db_persist = db_persist
-        super().__init__(**kwargs)
+        super().__init__(choices=getattr(output_field, "choices", None), **kwargs)
 
     @cached_property
     def cached_col(self):
```